### PR TITLE
adding potential fix for preds called by recursive preds

### PR DESCRIPTION
**Pull request by jyoo980**
See #34 for more

I'm not exactly confident with this fix, so a more stringent review would be appreciated.

Works for the minimal example that @ionathanch  provided, but on [`AVLTree.iterative.vpr`](https://github.com/jyoo980/silver/blob/master/src/test/resources/all/chalice/AVLTree.iterative.vpr), we still get an error (a different one downstream)

### CODE REVIEW

**Comment by jyoo980** [comment]
Don't review this, history got fucked

File: src/main/scala/viper/silver/plugin/standard/inline/InlineErrorChecker.scala [hunk 1]
```diff
@@ -43,6 +43,28 @@ trait InlineErrorChecker {
...
+    * Given a predicate, evaluate to a set containing the names of all predicates called by it.
+    * Only return the names of predicates called that aren't equal to itself.
+    *
+    * @param pred the predicate we want to collect the names of inner predicate calls for.
+    * @return a set of the ids of predicates called in the given predicate.
+    */
+  def nonRecursivePredsCalledBy(pred: Predicate): Option[Set[String]] =
+    pred.body.map { body =>
+      val children = body.subnodes
+      // Forgive me Father Alonzo for I have Sinned
```
Comment (ionathanch) [inline src/main/scala/viper/silver/plugin/standard/inline/InlineErrorChecker.scala:56]: :smiling_imp: :pray:

**Comment by ionathanch** [comment]
I was gonna say you'll want to update `cond` instead of passing another set of predicate names, that's what it's for

**Comment by jyoo980** [comment]
Is that safe though? `cond` is threaded through a lot of other places as well

**Comment by ionathanch** [comment]
`cond` is supposed to describe exactly which predicates you do want to touch
So updating it will fix not just unfold/fold but also unfolding
This also means that the predicates referred to by recursive ones won't be expanded, which we need if we aren't removing any of the unfolds
If you change the minimal example to this:
```
method nestedPred(this: Ref)
  requires F(this)
{
  unfold F(this)
  this.f := 0
}
```
The `unfold F(this)` will never get removed because it's erring on the safe side of not removing `F` at all (I'm assuming that's what we're doing), being referred to in `loop`, and to unfold `F` we need the precondition to remain `F` and not turn into `acc(this.f)`

TL;DR The answer is yes lol, only use `cond` for specifying predicates
